src/main/java/project_16x16/entities/Player.java

```diff
@@ -234,8 +230,8 @@ private void handleMouseInput() {
 			// Create Swing Projectile
 			swings.add(new Swing(applet, gameScene, (int) pos.x, (int) pos.y, state.facingDir));
 		}
-		for (int i = 0; i < swings.size(); i++) { // Update Swing Projectiles
-			swings.get(i).update();
+		for (Swing swing : swings) { // Update Swing Projectiles
+			swing.update();
```

Comment:
Reverting this fixes the `ConcurrentModificationException`.

Reply:
Strange that an enhanced for loop breaks the code... I'll revert the changes back to how it was!

Reply:
Yeah it is weird🤯

Reply:
Its caused by the fact that the `Swing` instance might get destroyed in `Swing::update()`:
```
if (animation.ended) {
	gameScene.getPlayer().swings.remove(this);
}
```
But I though foreach loops handeled destruction, perhaps there is a way to make it handle it?

Reply:
The reason the error gets thrown with an enhanced for-loop is explained [here](https://stackoverflow.com/a/9806502/9808792).

What's curious is how the concurrent modification is even happening (as far as I know there should be no possibility of concurrent code, but seemingly, there is).

Further, it seems the click from clicking the play button also gets registered in game (at the first frame) -- I wonder whether fixing this would stop the concurrent modification.